src/crimson/common/errorator.h

```diff
@@ -859,6 +860,45 @@ struct errorator {
     return make_ready_future<>();
   }
 
+  template <typename Iterator, typename Func>
+  static inline errorator<AllowedErrors...>::future<>
+  parallel_for_each(Iterator begin, Iterator end, Func&& func) noexcept {
+    parallel_for_each_state* s = nullptr;
+    // Process all elements, giving each future the following treatment:
+    //   - available, not failed: do nothing
+    //   - available, failed: collect exception in ex
+    //   - not available: collect in s (allocating it if needed)
+    while (begin != end) {
```

Comment:
nit, would be better to use `first` and `last` as the parameter names. to be more consistent with its C++ standard library counterpart of `std::for_each()`. in other words, the `begin()` and `end()` are function/method names, while `first` and `last` are parameters to define a range. could put
```c++
for (; first != last; ++first) {
```